synapse/storage/databases/main/registration.py

```diff
@@ -121,7 +121,7 @@ def __init__(
         database: DatabasePool,
         db_conn: LoggingDatabaseConnection,
         hs: "HomeServer",
-    ):
+    ) -> None:
```

Comment:
For interest: provided at least one other argument is typed, Mypy will allow you to omit the return type from `__init__`.

(I don't have strong feelings either way on this.)